### PR TITLE
[SIMP-7704] Update rule matching

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Fixed some bugs in the 'munge' portions of the native types.
- Improved the internal rule matching to handle most netmask and port updates.
- Added a `exact_match` Boolean to the `iptables_optimize` and
  `ip6tables_optimize` native types to allow for more aggressive rule matching.
  - This change requires that inbound rules match whatever is returned by
    `iptables-save` and/or `ip6tables-save` to prevent iptables flapping.

SIMP-7704 #close

### CODE REVIEW

**Comment by ayohrling** [comment]
I don't see how this spec test addresses SIMP-7704? I see the comment changes and verification, but no netmask change and verification.

**Comment by trevor-vaughan** [comment]
@ayohrling The Acceptance test updates handle the netmask flipping. The spec tests are lacking all around but this module is on its way out due to `firewalld`.

**Comment by trevor-vaughan** [comment]
After pulling this into additional tests for more modules, this is definitely a more correct default update. It does mean that we're going to need to be more careful about allowing SSH through on the beaker tests though :-D

File: REFERENCE.md [hunk 1]
```diff
@@ -48,8 +48,15 @@
...
 ### iptables
 
-Add management of iptables with default rule optimization and a failsafe
-fallback mode
+----------
+
+> It is **highly recommended** that you place this module in
+> ``firewalld`` mode if the underlying system supports it.
+>
+> You can do this by setting ``simp_options::firewall: firewalld` or
```
Comment (op-ct) [inline REFERENCE.md:56]: I'm leery of the `firewalld` value in the `$simp_options::firewall` catalyst, but that is another ticket

**Pull request merged**